packages/babel-plugin-transform-parameters/src/params.js

```diff
@@ -23,27 +22,27 @@ const buildSafeArgumentsAccess = template(`
   let $0 = arguments.length > $1 ? arguments[$1] : undefined;
 `);
 
-function isSafeBinding(scope, node) {
-  if (!scope.hasOwnBinding(node.name)) return true;
-  const { kind } = scope.getOwnBinding(node.name);
-  return kind === "param" || kind === "local";
-}
+const visitIdentifier = (path, state) => {
+  const { scope, node } = path;
+  const { name } = node;
+
+  if (
+    name === "eval" ||
+    (scope.getBinding(name) === state.scope.parent.getBinding(name) &&
+      state.scope.hasOwnBinding(name))
+  ) {
+    state.needsOuterBinding = true;
+    path.stop();
+  }
+};
 
 const iifeVisitor = {
-  ReferencedIdentifier(path, state) {
```

Comment:
I guess we can simplify to
```js
const iifeVisitor = {
  "ReferencedIdentifier|BindingIdentifier": visitIdentifier,
}
```
If there is a BindingIdentifier in the function scope, it should not satisfy `scope.getBinding(name) === state.scope.parent.getBinding(name)` so essentially it will be noop.